temporian/beam/io.py

```diff
@@ -67,6 +68,57 @@
 PEventSet = beam.PCollection[IndexValue]
 
 
+class UserEventSetFormat(Enum):
+    """Various representations of EventSets for user generation or consumption.
+
+    When combining Temporian program with your own beam stages, select the
+    most suited user event-set format compatible with your code.
+    """
+
+    eventKeyValue = "eventKeyValue"
+    """
+    Each event is represented as a dictionary of key to value for the features,
+    the indexes and the timestamp. Values are python primitives matching the
+    schema e.g. a `tp.int64` is feed as an `int`. Non-matching primitives are
+    casted e.g. a int is casted into a float permissively.
+
+    For example:
+        Schema
+            features=[("f1", DType.INT64), ("f2", DType.STRING)]
+            indexes=[("i1", DType.INT64), ("i2", DType.STRING)]
+
+        Data:
+            {"timestamp": 100.0, "f1": 1, "f2": b"a", "i1": 10, "i2": b"x"}
+    """
+
+    eventSetKeyValue = "eventSetKeyValue"
+    """
+    All the events in the same index are represented together in a dictionary of
+    key to values for the features, the indexes and the timestamp. Timestamps
+    are sorted. Values, index and timestamps are stored in Numpy arrays.
+
+    Index values are python primitives matching the schema e.g. a `tp.int64` is
+    feed as an `int`. Timestamps are a numpy array of float64. Feature values
+    are numpy array matching the schema e.g. a `tp.int64` is feed as a numpy
+    array of np.int64.
+
+    For example:
+
+        Schema
+            features=[("f1", DType.INT64), ("f2", DType.STRING)]
+            indexes=[("i1", DType.INT64), ("i2", DType.STRING)]
+
+        Data:
```

Comment:
For the same reason, i think this would be confusing.
*Adding "(one item)" to the documentation to make it clear that this is showing a single item (instead of the whole collection).